### ISSUE TITLE
Tuple enum variant generates unused_parens warning for Deserialize

### ISSUE THREAD

**Issue by compressed** [serde-rs/serde#1172]
After updating to `1.0.28` I started receiving an unused parens warning for an enum containing a tuple variant with at least two arguments (i.e. `A(usize)` won't trigger the warning). I don't get this warning on `1.0.27`.

Example:

```rust
extern crate serde;
#[macro_use]
extern crate serde_derive;

#[derive(Serialize, Deserialize)]
enum A {
    A(usize, usize),
}
```

Output:

```
warning: unnecessary parentheses around function argument
 --> src/lib.rs:5:21
  |
5 | #[derive(Serialize, Deserialize)]
  |                     ^^^^^^^^^^^ help: remove these parentheses
  |
  = note: #[warn(unused_parens)] on by default

warning: unnecessary parentheses around function argument
 --> src/lib.rs:5:21
  |
5 | #[derive(Serialize, Deserialize)]
  |                     ^^^^^^^^^^^ help: remove these parentheses

    Finished dev [unoptimized + debuginfo] target(s) in 0.19 secs
```

**Comment by dtolnay** [comment]
Thanks! The parentheses used to be a workaround for a compiler issue that has since been fixed: https://github.com/rust-lang/rust/issues/47311. I removed the workaround in 30361ac6d022839be19fc608645756ed1be839ab and released 1.0.29 which should have no warnings.

**Comment by compressed** [comment]
Thanks for the quick turnaround! No more warnings now :smile_cat:
